crates/shared/src/price_estimation/native/coingecko.rs

```diff
@@ -269,6 +271,27 @@ mod observe {
     }
 }
 
+mod metrics {
+    use {observe::metrics, prometheus::Histogram};
+
+    #[derive(prometheus_metric_storage::MetricStorage)]
+    pub(super) struct Metrics {
+        /// Tracks the CoinGecko batch size
+        #[metric(buckets(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20))]
+        coin_gecko_batch_size: Histogram,
+    }
+
+    impl Metrics {
+        fn get() -> &'static Self {
+            Metrics::instance(metrics::get_storage_registry()).unwrap()
+        }
+
+        pub(super) fn coin_gecko_batch_size(size: usize) {
+            Metrics::get().coin_gecko_batch_size.observe(size as f64);
```

Comment:
I think it's also possible to use a `GenericHistogram<AtomicUsize>` to not require casting here. Would also communicate more clearly what kind of values can be expected here.

Reply:
> umm I don't see anywhere a generic Histogram 🤔

Ahh, okay. `GenericCounter<T>` is a thing so I assumed `GenericHistogram<T>` as well but apparently not.

>What's more, this casting shouldn't be an issue since the code guarantees it isn't greater than MAX_BATCH_SIZE.

Was not really concerned about loss of precision but rather what the code communicates to the reader. But if `GenericHistogram<T>` is not a thing anyway this point becomes moot.